### ISSUE TITLE
Is there a way to declaratively copy over one file (not a dir) in the :transform key of a template.edn?

### ISSUE THREAD

**Issue by bombaywalla** [seancorfield/deps-new#17]
In trying to convert the `cryogen` template from `lein` to `deps-new`, I would like to copy over a few files (but not all) from a template dir to the new project dir.  The `:transform` key in `template.edn` seems to be for directories, not files.

**Comment by seancorfield** [comment]
Yes, it copies an entire directory, although it allows for special behavior for individual files so that they can be renamed and/or placed in different locations.

The idiom for `deps-new` templates is to use separate folders for each group of files that you want copied/processed together.

Can you elaborate on why you only want to copy part of a directory and not all of it, since you're creating a new template so you have control over the directories in it?

**Comment by bombaywalla** [comment]
Right.  If I had control over the template file structure, it would not be a problem.

The issue came up because Jakub Holý wanted (reasonably so) for the `deps-new` template to use the same files as the `lein` template.  The `lein` template files are set up as follows:

```
cryogen/
   deps.edn
   project.clj
   asc/
   md/
   config.edn
   themes/
```

and the project file structure should be:

```
deps.edn
project.clj
themes/
content/
   config.edn
   asc/
   md/
```

So, I was not able to use a `:transform` value to copy over `project.clj` and `deps.edn`.

I was wondering whether a `:transform` like `["src/blah.edn" "target/foo.edn"]` and/or `["src/blah.edn" "target"]` might be useful to add.

I can probably work around this particular issue with the cryogen `deps-new` template by modifying the `lein` template.  So, if you'd rather not add this capability at this time, I can still make progress.

**Comment by seancorfield** [comment]
I'll give it some thought but I might not get to it until the weekend.

`deps-new` is designed around you being in control of the file structure for the template -- and that's a big part of how it can be almost purely declarative. Starting with an "arbitrary" file structure and supporting "arbitrary" copying/filtering rules kind of runs counter to the whole design.

My reaction to seeing the structure above is that the `lein` template is poorly structured -- but I know a lot of `lein` templates are "flat" because you _have to write code_ to process everything so why bother making it "structured", eh?

**Comment by bombaywalla** [comment]
No hurry from my side.  It was just a suggestion for a possible enhancement.

Agreed on the `lein` template structure.  In the case of the `cryogen` template, I'm going modify it to make it more structured.  It will even make the `lein` template code more consistent.

And, I understand what you're saying about the declarative design of `deps-new`.

**If** you are considering adding something for individual files (in contrast to directories) then another option is to add an `:only` specification (perhaps just before the `:raw` specification).

So, `{:transform [["src" "target" {"file1" "file1x" "file2" "file2x"} :only]]}` would mean: Copy **ONLY** the `file1` and `file2` from `src` dir to `target` dir and optionally rename (with or without substitutions).

**Comment by seancorfield** [comment]
Yeah, that's close to what I am circling around...

**Comment by seancorfield** [comment]
I experimented with two approaches and decided to go with your suggestion of `:only` so that is partially implemented on **develop**

**Comment by seancorfield** [comment]
Implemented in v0.4.3.
